### PR TITLE
[FIX #23] Added new twitter widget using twitter post fetcher

### LINKED ISSUE #23: Fixing the vacant space around twitter widget
Current implementation of the twitter widget leaves unnecessary space to the left and right also it doesn't go with the theme. A solution is required by going through this comment on the original issue -> [Here](https://github.com/pythonindia/inpycon2017/issues/18#issuecomment-313901135)

### PR DESCRIPTION
**Pull request by amitsin6h**
Fixed the vacant space around twitter widget using https://github.com/jasonmayes/Twitter-Post-Fetcher.
Please review and let me know if some changes are required in css for designing the layout.

### CODE REVIEW

**Comment by CuriousLearner** [comment]
Hey,

Thanks for your contributions.

This breaks on small screens. You need to handle for changes for all device less than `md` size.

Also, about the `reply`, `retweet`, `Favourite` shall be converted to buttons or something else. This doesn't look good in a link form.

**Comment by amitsin6h** [comment]
@CuriousLearner ok thanks for your suggestion.

**Comment by amitsin6h** [comment]
Old Preview of twitter latest post
old-preview


New Preview of twitter latest post. Hope everyone like this also and also it's also responsive
new-preview

**Review by realslimshanky — APPROVED** [review]
LGTM.
Over to you @CuriousLearner

**Comment by CuriousLearner** [comment]
@amitsin6h Thanks for your work. Can you please rebase your changes with the current HEAD at master?

You can do this while pulling from PyCon India repo like:

`git pull git@github.com:pythonindia/inpycon2017.git  master --rebase`

Then resolve conflicts if any, and regenerate the files again :) Let me know here if you need help.

**Comment by amitsin6h** [comment]
@CuriousLearner I have made all the changes. Please review it

File: coc.html [hunk 1]
```diff
@@ -200,6 +201,9 @@ <h1 class="animated fadeInDown">Code Of Conduct</h1>
 	<!--************************************
 			Wrapper End
 	*************************************-->
+	<script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/moment.js/2.8.3/moment.min.js"></script>
+    <script type="text/javascript" src="js/vendor/twitter-post-fetcher.js"></script>
```
Comment (sayanchowdhury) [inline coc.html]: Indentation.

File: coc.html [hunk 2]
```diff
@@ -200,6 +201,9 @@ <h1 class="animated fadeInDown">Code Of Conduct</h1>
...
+    <script type="text/javascript" src="js/vendor/twitter-post-fetcher-config.js"></script>
```
Comment (sayanchowdhury) [inline coc.html]: Indentation

File: index.html [hunk 3]
```diff
@@ -2581,7 +2582,7 @@ <h3 class="wow fadeInDown">Sponsors</h3>
 			<!--************************************
 					Locations End
 			*************************************-->
-			<!--************************************
+						<!--************************************
```
Comment (sayanchowdhury) [inline index.html]: There should be no blank spaces.

File: index.html [hunk 4]
```diff
@@ -2591,107 +2592,17 @@ <h3 class="wow fadeInDown">Sponsors</h3>
...
-										<div class="tg-posttitle">
-											<h3><a href="">All That Glitters Is Not Gold</a></h3>
-										</div>
-										<div class="tg-description">
-											<p>Consectetur adipisicing elit sed eiusmod tempor inciunt labore dolore magna aliqua enim.</p>
-										</div>
-									</div>
-								</article>
+							<div class="col-sm-12 col-md-12 col-lg-12">
+								<div id="pyconindiaTweets"></div>
```
Comment (sayanchowdhury) [inline index.html]: Can we have a better id here, separated by hyphens `-`

File: js/vendor/twitter-post-fetcher-config.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+/*Twitter Post Fetcher config*/
+
+var configProfile = {
+  "profile": {"screenName": 'pyconindia'},
+  "domId": 'pyconindiaTweets',
```
Comment (sayanchowdhury) [inline js/vendor/twitter-post-fetcher-config.js]: This would change accordingly.

**Review by sayanchowdhury — CHANGES_REQUESTED** [review]

File: index.html (hunk 4 above)
Comment (CuriousLearner) [reply index.html]: Also, please keep them lower case separated by hyphens

**Comment by sayanchowdhury** [comment]
You need to rebase the branch on top of `master` again

**Review by amitsin6h — COMMENTED** [review]
please review made the changes

**Comment by realslimshanky** [comment]
There's a couple of UI bugs you need to fix on your code.
- [X] The twitter feed is breaking the page on mobile view due to text overflow.
* Exhibit A 
Exhibit A
* Exhibit B
Exhibit B

Possible Fix -
* Wrap text inside `tweet` div
* Make text overflow on the `tweet` div inside `timeline-content` div

- [x] `Reply` button is little of towards right
* Exhibit A
Exhibit A

Possible Fix -
* Try `ul > li` combination.

**Review by amitsin6h — COMMENTED** [review]
@realslimshanky please review it

**Comment by realslimshanky** [comment]
Please check the buttons. The top one is still not aligned properly.

**Review by amitsin6h — COMMENTED** [review]
@realslimshanky fixed the top button. I tried to reduce the radius but still it didn't worked so I changed the view in the small screen. Let me know once you review it

File: templates/sections/_tweets.html [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+							<div class="col-sm-12 col-md-12 col-lg-12">
+								<div id="pyconindia-tweets"></div>
+							</div>
+						</div>
+					</div>
+				</div>
+			</section>
+			<!--************************************
+					Latest Tweet End
+			*************************************-->
```
Comment (sayanchowdhury) [inline templates/sections/_tweets.html]: The indentation of the files are 2 spaces now.

**Review by sayanchowdhury — CHANGES_REQUESTED** [review]

**Review by realslimshanky — CHANGES_REQUESTED** [review]
Instead of a separate file for css you should add custom css inside main.css.
Your 'Twitter Post Fetcher' should go between 'Speakers Style' and 'Venue Style'.
Other than @sayanchowdhury 's review and this everything else LGTM.

**Review by amitsin6h — COMMENTED** [review]
@sayanchowdhury @realslimshanky Please review it.

**Review by realslimshanky — APPROVED** [review]
LGTM

**Review by sayanchowdhury — APPROVED** [review]

**Review by CuriousLearner — CHANGES_REQUESTED** [review]
Functionality wise, "LGTM" 

Please fix indentation in CSS and HTML file. Please use 2 spaces indentation to keep it consistent with the codebase. 

Once you're done, let me know and I'll merge it!

Thanks for your contribution!

**Comment by amitsin6h** [comment]
but I did the same as the other html and css file was done. I mean the indentation @CuriousLearner 
let me check again

**Review by amitsin6h — COMMENTED** [review]
@CuriousLearner please review the indent

**Comment by amitsin6h** [comment]
In my sublime text editor everything looks great but I don't know why I get indentation problem.

**Comment by amitsin6h** [comment]
@CuriousLearner

**Comment by CuriousLearner** [comment]
@amitsin6h Apologies for getting to this late. 

Please use 2 spaces indentation (You're using tab for indentation. In sublime text editor select to spaces as indentation and keep tab size as 2).

And please do that only to files in templates folder and the css source files. Then you've to regenerate the html.

I hope that makes sense.

**Review by amitsin6h — COMMENTED** [review]
@CuriousLearner Please review it now. Used tab_size:2 in sublime. Hope this works

File: css/bootstrap.min.css [hunk 7]
```diff
@@ -2,5 +2,5 @@
...
...osition:relative;font-size:0;white-space:nowrap}.input-group-btn>.btn{position:relative}.input-group-btn>.btn+.btn{margin-left:-1px}.input-group-btn>.btn:active,.input-group-btn>.btn:focus,.input-group-btn>.btn:hover{z-index:2}.input-group-btn:first-child>.btn,.input-group-btn:first-child>.btn-group{margin-right:-1px}.input-group-btn:last-child>.btn,.input-group-btn:last-child>.btn-group{z-index:2;margin-left:-1px}.nav{padding-left:0;margin-bottom:0;list-style:none}.nav>li{position:relative;display:block}.nav>li>a{position:relative;display:block;padding:10px 15px}.nav>li>a:focus,.nav>li>a:hover{text-decoration:none;background-color:#eee}.nav>li.disabled>a{color:#777}.nav>li.disabled>a:focus,.nav>li.disabled>a:hover{color:#777;text-decoration:none;cursor:not-allowed;background-color:transparent}.nav .open>a,.nav .open>a:focus,.nav .open>a:hover{background-color:#eee;border-color:#337ab7}.nav .nav-divider{height:1px;margin:9px 0;overflow:hidden;background-color:#e5e5e5}.nav>li>a>img{max-width:none}.nav-tabs{border-bottom:1px solid #ddd}.nav-tabs>li{float:left;margin-bottom:-1px}.nav-tabs>li>a{margin-right:2px;line-height:1.42857143;border:1px solid transparent;border-radius:4px 4px 0
```
Comment (CuriousLearner) [inline css/bootstrap.min.css]: No need to touch default bootstrap file.

File: templates/sections/_tweets.html [hunk 8]
```diff
@@ -0,0 +1,28 @@
+<!--************************************
+  Latest Tweet Start
+*************************************-->
+<section class="tg-sectionspace tg-haslayout">
+  <div class="container">
+    <div class="row">
+    	<div class="col-xs-offset-0 col-xs-12 col-sm-offset-0 col-sm-12 col-md-offset-2 col-md-8 col-lg-offset-2 col-lg-8">
+    		<div class="tg-sectionhead">
```
Comment (CuriousLearner) [inline templates/sections/_tweets.html:8]: 2 spaces indentation here as well (for the entire nested div)

File: templates/index.html [hunk 9]
```diff
@@ -2,2417 +2,2300 @@
...
-                                  <span class="tg-eventcatagory">Senior manager</span>
-                                </div>
-                              </div>
-                            </div>
-                          </div>
-                          <div class="tg-event">
-                            <div class="tg-eventspeaker">
-                              <figure class="tg-eventspeakerimg tg-themeimgborder">
-                                <img src="images/speakers/img-08.jpg" alt="image description">
-
```
Comment (CuriousLearner) [inline templates/index.html]: Indentation of this file is altered from default 2 space. Please fix.

File: templates/sections/_tweets.html [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    		<div class="col-sm-12 col-md-12 col-lg-12">
+    			<div id="pyconindia-tweets"></div>
+    		</div>
+    	</div>
+    </div>
+  </div>
+</section>
+<!--************************************
+  Latest Tweet End
+*************************************-->
```
Comment (CuriousLearner) [inline templates/sections/_tweets.html]: Leave a blank line at the end of the file.

**Review by CuriousLearner — CHANGES_REQUESTED** [review]

File: css/bootstrap.min.css (hunk 7 above)
Comment (amitsin6h) [reply css/bootstrap.min.css]: the default bootstrap was creating problem in alignment of button in small screen so made some changes. but those changes does not affect the website. other than solving button alignment issue.

Comment (CuriousLearner) [reply css/bootstrap.min.css]: If it is posing a problem, you're supposed to override style in your custom css file.

**Review by amitsin6h — COMMENTED** [review]
@CuriousLearner Please review it.

File: templates/sections/_tweets.html [hunk 11]
```diff
@@ -6,7 +6,7 @@
     <div class="row">
     	<div class="col-xs-offset-0 col-xs-12 col-sm-offset-0 col-sm-12 col-md-offset-2 col-md-8 col-lg-offset-2 col-lg-8">
     		<div class="tg-sectionhead">
-    			<div class="tg-sectionheading">
+    		  <div class="tg-sectionheading">
     				<h2>Always Stay Updated</h2>
```
Comment (CuriousLearner) [inline templates/sections/_tweets.html]: all the lines below.

Everything indented with 2 spaces.

**Comment by amitsin6h** [comment]
#108 Please review it. Closing this PR

**Pull request closed without merging**